### DISCUSSION TITLE
How to remove the extra domains?

### DISCUSSION THREAD

**Discussion by madzadev** [vercel/vercel#5711] [Help]
I deployed an app from Github.
Vercel gave me 3 domains:

https://project-name-m3jh45hf.vercel.app
https://project-name-git-master.user.vercel.app/
https://project-name.user.vercel.app/

After that, I added a custom domain https://mysite.com, hoping the 3 other domains would be removed, but they are still active and now my site is accessible through 4 different links.

How to remove the other three, leaving just https://mysite.com?

**Comment by jacobhq** [comment]
That's not possible at the moment.

**Reply by madzadev** [reply]
Anyone can elaborate on the reasons behind this? 👀
Most platforms I know give you just a single link per deployed project 😉 Do not fully understand why someone would need 4, without an option to remove any of them 🤔

**Reply by wadamek65** [reply]
Would also like to know 😕 As far as I know you are able to remove all deployment aliases generated by vercel except one. Running `vercel alias rm` for the last one gives an alias not found error 😕

**Reply by jacobhq** [reply]
I think you get a domain for each project, a domain for each deployment￼￼, and a domain that is at the head of every branch￼.

**Reply by fu-sen** [reply]
Vercel was originally designed to be developed by multiple people so that it can be checked before it is released.
So the public CLI command is `vercel --prod` instead of` vercel`.

Therefore, deletion is not provided, but you can make it Redirect to make sure the domain is displayed:
https://vercel.com/docs/custom-domains#redirecting-domains

Also, the `<link rel=”canonical”>` tag is an effective way for Google and others to recognize the correct address and display the search results.

**Comment by kuangthien** [comment]
The preview domains even appear on a lot of indexing sites such as Alexa, I dont'  believe in privacy policies of vercel anymore . Now I even cannot revert to old domains to show a 404 or add a replaced robot.txt for my old preview domains.

I love Next, but Vercel....

**Reply by fu-sen** [reply]
Really?
https://www.google.com/search?q=site%3Avercel.app

This is a subdomain exposed by `vercel --prod`. **You can use `SUBDOMAIN.vercel.app` instead of custom domain.**
Do you see `EXAMPLE-USER.vercel.app` in your search results?

**Reply by kuangthien** [reply]
I see PROJECT_NAME-HASH_STRING-**MY_USERNAME**.vercel.app on Alexa
It's the preview deploy link
Now I cannot remove it on Alexa. Unbelievable
Anybody from Vercel can pls help me... :(

**Reply by paulogdm** [reply]
Please send us a message through our support channel: https://vercel.com/support/request

**Reply by kuangthien** [reply]
Thx you, I submitted #00018984 , But I don't know it's scope of Vercel supports or not. I tend to buy another production domain if the problem cannot solve.
Unbelievable
